lib/generator.ts

```diff
@@ -597,14 +702,35 @@ export class Generator extends Cli {
       rootType: IBasePathType.root,
       noExistenceCheck: true,
     })
+    return mainFile
+  }
 
-    source +=
-      `\ndeclare module '${packageDetails.name}' {\n` +
-      `  import main = require('${mainFile}');\n` +
-      '  export = main;\n' +
-      '}'
+  /**
+   * Append template where {0} is replaced with the name/path of the entry module.
+   * @param source generated .d.ts declaration source so far
+   */
+  private addTemplate(source: string) {
+    const tpl = this.getTemplate()
+    if (!tpl) return source
+    verbose('Adding template')
 
-    verbose('Successfully created alias for main file!')
+    const entry = this.getEntry()
+
+    if (!entry) {
+      error('No entry file is available!')
+      throw new Error('No entry file is available!')
+    }
+
+    const mainFile = this.convertPathToModule(resolve(this.getRoot(), entry), {
+      rootType: IBasePathType.root,
+      noExistenceCheck: true,
+    })
+
+    const appliedTemplate = tpl.replace('{0}', mainFile)
+
+    source += '\n' + appliedTemplate + '\n'
+
+    verbose('Successfully added template!')
```

Comment:
Test coverage for all new functionality would be good. During current review I saw a few places where code could break.